File: src/server/python/clock.py
class LamportClock:
    def __init__(self):
        self.time = 0

    def tick(self):
        """Incrementa o relógio antes de enviar uma mensagem"""
        self.time += 1
        return self.time

    def update(self, received_time: int):
        """Atualiza o relógio ao receber uma mensagem"""
        self.time = max(self.time, received_time) + 1
        return self.time

    def get_time(self):
        """Retorna o tempo atual sem modificar"""
        return self.time

File: src/bot/node/clock.js
class LamportClock {
    constructor() {
        this.time = 0;
    }

    tick() {
        this.time += 1;
        return this.time;
    }

    update(receivedTime) {
        this.time = Math.max(this.time, receivedTime) + 1;
        return this.time;
    }

    getTime() {
        return this.time;
    }
}

module.exports = LamportClock;
